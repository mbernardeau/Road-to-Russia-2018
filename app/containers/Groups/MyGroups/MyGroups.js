import React from 'react';
import PropTypes from 'prop-types';

import {
  map,
} from 'lodash';

import Card from 'material-ui/Card';
import Table, { TableBody } from 'material-ui/Table';

import GroupRow from './GroupRow';

const MyGroups = ({ groups }) => (
  <Card style={styles.container}>
    <h2>Mes groupes</h2>
    <Table>
      <TableBody>
        {map(groups, (g, key) => (
          <GroupRow group={g} key={key} />
        ))}
      </TableBody>
    </Table>
  </Card>
);


const styles = {
  container: {
    width: '90%',
  },
};

MyGroups.propTypes = {
  groups: PropTypes.objectOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
    })
  ),
};

export default MyGroups;
