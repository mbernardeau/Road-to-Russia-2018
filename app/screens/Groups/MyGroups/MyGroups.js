import React from 'react'
import PropTypes from 'prop-types'

import { map, isEmpty } from 'lodash'

import Card from 'material-ui/Card'
import Table from 'material-ui/Table'
import TableBody from 'material-ui/Table/TableBody'
import Typography from 'material-ui/Typography'

import GroupRow from './GroupRow'

const MyGroups = ({ groups }) =>
  isEmpty(groups) ? null : (
    <Card style={styles.container}>
      <Typography gutterBottom type="headline">
        Mes groupes
      </Typography>

      <Table>
        <TableBody>{map(groups, (g, key) => <GroupRow group={g} key={key} />)}</TableBody>
      </Table>
    </Card>
  )

const styles = {
  container: {
    width: '90%',
    paddingTop: 15,
  },
}

MyGroups.propTypes = {
  groups: PropTypes.objectOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
    }),
  ),
}

export default MyGroups
