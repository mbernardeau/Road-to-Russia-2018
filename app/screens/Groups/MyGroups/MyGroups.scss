.my-groups-card {
  padding: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
}
