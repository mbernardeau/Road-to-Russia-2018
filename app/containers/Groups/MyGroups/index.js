import MyGroups from './MyGroupsContainer';

export default MyGroups;
