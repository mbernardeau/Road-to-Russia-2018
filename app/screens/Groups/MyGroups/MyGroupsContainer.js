import { firebaseConnect } from 'react-redux-firebase'
import { connect } from 'react-redux'
import { compose } from 'redux'

import { getUserGroups } from 'redux/groups'

import MyGroups from './MyGroups'

export default compose(
  firebaseConnect([{ path: 'groups' }]),
  connect(state => ({
    groups: getUserGroups(state),
  })),
)(MyGroups)
