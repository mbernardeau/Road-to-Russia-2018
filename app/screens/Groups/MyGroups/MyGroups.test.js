import React from 'react'
import renderer from 'react-test-renderer'

import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'

import MyGroups from './MyGroups'

jest.mock('material-ui/Card', () => 'Card')
jest.mock('material-ui/Table', () => 'Table')
jest.mock('material-ui/Table/TableBody', () => 'TableBody')
jest.mock('material-ui/Typography', () => 'Typography')
jest.mock('./GroupRow', () => 'GroupRow')

describe('Groups/MyGroups', () => {
  const state = {
    firebase: {
      auth: {
        uid: 'iduser',
      },
    },
  }

  const store = configureMockStore()(state)

  it('renders correctly when no group joined', () => {
    const props = {
      groups: {},
    }

    const tree = renderer
      .create(
        <Provider store={store}>
          <MyGroups {...props} />
        </Provider>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with different groups', () => {
    const props = {
      groups: {
        idGroup: {
          name: 'my-test-group',
          members: {
            iduser: 'member',
          },
        },
        idGroup2: {
          name: 'my-test-group-2',
          members: {
            iduser: 'awaiting',
          },
        },
        idGroup3: {
          name: 'my-test-group-3',
          members: {
            iduser: 'admin',
          },
        },
      },
    }

    const tree = renderer
      .create(
        <Provider store={store}>
          <MyGroups {...props} />
        </Provider>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
