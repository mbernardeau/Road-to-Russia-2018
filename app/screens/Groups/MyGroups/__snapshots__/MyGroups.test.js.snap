// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Groups/MyGroups renders correctly when no group joined 1`] = `null`;

exports[`Groups/MyGroups renders correctly with different groups 1`] = `
<Card
  style={
    Object {
      "paddingTop": 15,
      "width": "90%",
    }
  }
>
  <Typography
    gutterBottom={true}
    type="headline"
  >
    Mes groupes
  </Typography>
  <Table>
    <TableBody>
      <GroupRow
        group={
          Object {
            "members": Object {
              "iduser": "member",
            },
            "name": "my-test-group",
          }
        }
      />
      <GroupRow
        group={
          Object {
            "members": Object {
              "iduser": "awaiting",
            },
            "name": "my-test-group-2",
          }
        }
      />
      <GroupRow
        group={
          Object {
            "members": Object {
              "iduser": "admin",
            },
            "name": "my-test-group-3",
          }
        }
      />
    </TableBody>
  </Table>
</Card>
`;
